proxy_manager.go

```diff
@@ -231,3 +231,16 @@ func (m *proxyManager) getFlakeIDGeneratorConfig(name string) FlakeIDGeneratorCo
 func makeProxyName(serviceName string, objectName string) string {
 	return fmt.Sprintf("%s%s", serviceName, objectName)
 }
+
+type proxyDestroyer interface {
+	Destroy(ctx context.Context) error
+}
+
+func (m *proxyManager) destroyProxies(ctx context.Context) {
+	for key, p := range m.proxies {
```

Comment:
`proxies` are protected with a mutex. That should be used in this function before accessing that.

Reply:
Resolved, [0042502](https://github.com/hazelcast/hazelcast-go-client/pull/772/commits/0042502d165af4bbfd57b99474f816dd147c4599).